spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/FilteringMessageListenerAdapter.java

```diff
@@ -41,13 +47,69 @@
 	public FilteringMessageListenerAdapter(MessageListener<K, V> delegate,
 			RecordFilterStrategy<K, V> recordFilterStrategy) {
 		super(delegate, recordFilterStrategy);
+		this.ackDiscarded = false;
+	}
+
+	/**
+	 * Create an instance with the supplied strategy and delegate listener.
+	 * @param delegate the delegate.
+	 * @param recordFilterStrategy the filter.
+	 * @param ackDiscarded true to ack (commit offset for) discarded messages when the
+	 * listener is configured for manual acks.
+	 */
+	public FilteringMessageListenerAdapter(MessageListener<K, V> delegate,
+			RecordFilterStrategy<K, V> recordFilterStrategy, boolean ackDiscarded) {
+		super(delegate, recordFilterStrategy);
+		this.ackDiscarded = ackDiscarded;
 	}
 
 	@Override
-	public void onMessage(ConsumerRecord<K, V> consumerRecord) {
+	public void onMessage(ConsumerRecord<K, V> consumerRecord, Acknowledgment acknowledgment, Consumer<?, ?> consumer) {
 		if (!filter(consumerRecord)) {
-			this.delegate.onMessage(consumerRecord);
+			switch (this.delegateType) {
+				case ACKNOWLEDGING_CONSUMER_AWARE:
+					this.delegate.onMessage(consumerRecord, acknowledgment, consumer);
+					break;
+				case ACKNOWLEDGING:
+					this.delegate.onMessage(consumerRecord, acknowledgment);
+					break;
+				case CONSUMER_AWARE:
+					this.delegate.onMessage(consumerRecord, consumer);
+					break;
+				case SIMPLE:
+					this.delegate.onMessage(consumerRecord);
+			}
+			switch (this.delegateType) {
```

Comment:
Missed `else` around this switch . 
Although I'd say we don't need switch at all. One `if` should be enough:
```
		else {
			if (this.ackDiscarded && acknowledgment != null &&
					(ListenerType.ACKNOWLEDGING_CONSUMER_AWARE.equals(this.delegateType) ||
							ListenerType.ACKNOWLEDGING.equals(this.delegateType))) {
				acknowledgment.acknowledge();
			}
		}

```